src/google-maps/google-map/google-map.ts

```diff
@@ -464,9 +464,9 @@ export class GoogleMap implements OnChanges, OnInit, OnDestroy {
       ...options,
       // It's important that we set **some** kind of `center` and `zoom`, otherwise
       // Google Maps will render a blank rectangle which looks broken.
-      center: this._center || options.center || DEFAULT_OPTIONS.center,
-      zoom: this._zoom ?? options.zoom ?? DEFAULT_OPTIONS.zoom,
-      mapTypeId: this.mapTypeId || options.mapTypeId
+      center: this._center || this.googleMap?.getCenter() || options.center || DEFAULT_OPTIONS.center,
```

Comment:
Shouldn't the `options.center` take precedence over the one from `getCenter`? Otherwise if the map already has a center it'll always override the one from the options. Also we should have a comment about why it works this way and some unit tests.

Reply:
I think not, the order you see now, is the same as it was. setting `center` via `@Input` sets `this._center` while `this._options` is still old variant. I don't think this approach works in all situations.

I would think there should be 2 methods for `combineOptions`, one that updates options based on new settings object, and one that overrides per-key basis...